server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -602,6 +602,14 @@ public MappedFieldType fieldType(String fullName) {
         return fieldTypes.simpleMatchToFullName(pattern);
     }
 
+    /**
+     * Given a field name, returns its possible paths in the _source. For example,
+     * the 'source path' for a multi-field is the path to its parent field.
+     */
+    public Set<String> sourcePath(String fullName) {
```

Comment:
I wonder if this should return something that has an `extractValue(SourceLookup)` method on it. I'm just thinking of types `constant_keyword` and the formatting stuff we talked about over video. Not that now is the time to build that, but maybe this is the place to handle it.

Reply:
Will keep this in mind in the next PR where we'll handle field-specific behavior like formatting.

Reply:
:+1: